internal/cmd/plugin/logical/publication/create/cmd.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/logical"
+)
+
+// NewCmd initializes the publication create command
+func NewCmd() *cobra.Command {
+	var dbName string
+	var allTables bool
+	var schemaNames []string
+	var tableExprs []string
+	var publicationName string
+	var externalClusterName string
+	var dryRun bool
+
+	publicationCreateCmd := &cobra.Command{
+		Use:   "create cluster_name",
+		Args:  cobra.ExactArgs(1),
+		Short: "create a logical replication publication",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dbName := strings.TrimSpace(dbName)
+			publicationName := strings.TrimSpace(publicationName)
+			clusterName := args[0]
+
+			if len(dbName) == 0 {
+				var err error
+				dbName, err = logical.GetApplicationDatabaseName(cmd.Context(), clusterName)
+				if err != nil {
+					return err
+				}
+			}
+			if len(dbName) == 0 {
+				return fmt.Errorf(
+					"the name of the database was not specified and there is no available application database")
+			}
+
+			if len(publicationName) == 0 {
+				return fmt.Errorf("publication is a required option")
+			}
+
+			if allTables && (len(schemaNames) > 0 || len(tableExprs) > 0) {
+				return fmt.Errorf("cannot publicate all tables and selected schema/tables at the same time")
+			}
+			if !allTables && len(schemaNames) == 0 && len(tableExprs) == 0 {
+				return fmt.Errorf("no publication target selected")
+			}
```

Comment:
We could delegate this to cobra, using `MarkFlagRequired`, `MarkFlagsMutuallyExclusive` and `MarkFlagsOneRequired`. See https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#required-flags